packages/block-library/src/query/edit/query-content.js

```diff
@@ -46,10 +46,14 @@ export default function QueryContent( {
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: TEMPLATE,
 	} );
-	const { postsPerPage } = useSelect( ( select ) => {
+	const { postsPerPage, currentPostType } = useSelect( ( select ) => {
 		const { getSettings } = select( blockEditorStore );
 		const { getEntityRecord, getEntityRecordEdits, canUser } =
 			select( coreStore );
+		// @wordpress/block-library should not depend on @wordpress/editor.
+		// However, we need to get the current post type in order to inherit the correct query.
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		const { getCurrentPostType } = select( 'core/editor' );
```

Comment:
If we need to access the current postType we should be able to get it via the block context by setting `usesContext` to include `postType`.

However I'm not sure this approach is solving the problem described 🤔 

Instead of showing the posts of the current post type I do think the default state for the inherit option should still be that is always renders posts of the posts posts type. 

The issue is that today it doesn't do that when the query loop is inserted inside content rather than inside a template. 

Rather than this I think it might be a better route to add a check in PHP to check whether something like whether the inherit option is set to true and the `is_singular` check resolves to true. 

In that case we change the default params to set the post type to `post`

Reply:
Thanks @fabiankaegy! 

> If we need to access the current postType we should be able to get it via the block context by setting usesContext to include postType.

Ah thank you, that's helpful.

> Instead of showing the posts of the current post type I do think the default state for the inherit option should still be that is always renders posts of the posts posts type.

I agree - I was focused on making the Editor and the front end consistent, and assumed the front end was correct. But now I think it makes more sense for the query loop to default to querying posts, as this seems to be the safest assumption for most users. I've tried out `is_singular()` here: https://github.com/WordPress/gutenberg/pull/65067/commits/51f93838592dc2456332967e847985985b1f7bd1

This new behavior also matches the block description:

![image](https://github.com/user-attachments/assets/16dc16b2-8e2d-4754-8e74-645c9a6e24d8)